packages/web/src/pages/upload-page/fields/TrackMetadataFields.tsx

```diff
@@ -0,0 +1,135 @@
+import { useState } from 'react'
+
+import { GENRES, getCanonicalName, FeatureFlags } from '@audius/common'
+import { useField } from 'formik'
+
+import DropdownInput from 'components/data-entry/DropdownInput'
+import Input from 'components/data-entry/Input'
+import TagInput from 'components/data-entry/TagInput'
+import TextArea from 'components/data-entry/TextArea'
+import UploadArtwork from 'components/upload/UploadArtwork'
+import { getFeatureEnabled } from 'services/remote-config/featureFlagHelpers'
+import { resizeImage } from 'utils/imageProcessingUtil'
+import { moodMap as untypedMoodMap } from 'utils/moods'
+
+import styles from './TrackMetadataFields.module.css'
+
+const moodMap: Record<string, JSX.Element> = untypedMoodMap
+const MOODS = Object.keys(moodMap).map((k) => ({
+  text: k,
+  el: moodMap[k]
+}))
+
+const messages = {
+  genre: 'Pick a Genre'
+}
+
+type TrackMetadataFieldsProps = {
+  /** Whether or not the preview is playing. */
+  playing: boolean
+  // type: 'track' | 'album' | 'playlist'
+  type: 'track'
+}
+
+const TrackMetadataFields = (props: TrackMetadataFieldsProps) => {
+  const [imageProcessingError, setImageProcessingError] = useState(false)
+  const [artworkField, , artworkHelpers] = useField('artwork')
+  const [titleField, titleMeta, titleHelpers] = useField('title')
+  const [genreField, genreMeta, genreHelpers] = useField('genre')
+  const [, moodMeta, moodHelpers] = useField('mood')
+  const [, tagsMeta, tagsHelpers] = useField('tags')
+  const [, descriptionMeta, descriptionHelpers] = useField('description')
+
+  const onDropArtwork = async (selectedFiles: File[], source: string) => {
+    try {
+      let file = selectedFiles[0]
+      file = await resizeImage(file)
+      const storageV2SignupEnabled = getFeatureEnabled(
+        FeatureFlags.STORAGE_V2_SIGNUP
+      )
+      const storageV2UploadEnabled = getFeatureEnabled(
+        FeatureFlags.STORAGE_V2_TRACK_UPLOAD
+      )
+      if (storageV2SignupEnabled || storageV2UploadEnabled) {
+        // @ts-ignore
+        file.name = selectedFiles[0].name
```

Comment:
does casting to string not work? ts ignore is a bit scary

Reply:
Unfortunately, no. The type error is that `file.name` is a readonly field. This was copied from an untyped js file https://github.com/AudiusProject/audius-client/blob/main/packages/web/src/components/data-entry/FormTile.js#L842
I didn't want to break the logic, but I'll follow up with Theo to see if it's actually working as intended.